### PR TITLE
Adding models

### PR DESCRIPTION
**Pull request by josephignace**
Neuro model.

### CODE REVIEW

File: analysis/active_analyses.R
Comment (venexia) [inline analysis/active_analyses.R:74]: These covariates are not listed in the protocol and should not be included in the model.

File: analysis/active_analyses.R
Comment (venexia) [inline analysis/active_analyses.R:531]: Please use `all_covars` for consistency.

File: analysis/active_analyses.R
Comment (venexia) [inline analysis/active_analyses.R:554]: Please use `all_covars` for consistency.

File: analysis/model/make_model_input.R
Comment (venexia) [inline analysis/model/make_model_input.R:118]: The variables used for filtering should be removed at this stage as they now contain no information and are just making the dataset bigger.

File: analysis/model/make_model_input.R
Comment (venexia) [inline analysis/model/make_model_input.R:490]: This subgroup should be defined by history of Parkinson's disease rather than history of any dementia.

File: analysis/model/make_model_input.R
Comment (venexia) [inline analysis/model/make_model_input.R:508]: This subgroup should be defined by history of Parkinson's disease rather than history of any dementia.

File: analysis/model/make_model_input.R
Comment (venexia) [inline analysis/model/make_model_input.R:522]: This comment applies to all new subgroups: Please could you rename the subgroups in active analyses so they are consistent with the variables you are using - this makes it easier for people unfamiliar with the project to track what is going on. For example, the subgroup variable here is called "sub_prior_parkinson_risk_true" and it uses the variable "cov_bin_history_parkinson_risk". I suggest we use "sub_history_parkinson_risk_true" to match "cov_bin_history_parkinson_risk".

**Review by venexia — CHANGES_REQUESTED** [review]
HI @josephignace. Thank you for submitting this PR. Some small changes are required for consistency and there are a couple of extra covariates to be removed from the active analyses table before I can approve this request. I am also just checking it runs locally so will confirm whether it works fully later today.

File: analysis/active_analyses.R
Comment (josephignace) [reply analysis/active_analyses.R]: Removed covariates that are not listed in the protocol. The rest are listed in the protocol (similarly to MH and GI).

File: analysis/active_analyses.R
Comment (josephignace) [reply analysis/active_analyses.R]: Done.

File: analysis/model/make_model_input.R
Comment (josephignace) [reply analysis/model/make_model_input.R]: Removing the variables, remove covariates from the models (listed in the protocol). Dropping columns or not, produce files of the same size.

File: analysis/model/make_model_input.R
Comment (josephignace) [reply analysis/model/make_model_input.R]: Done.

File: analysis/model/make_model_input.R
Comment (josephignace) [reply analysis/model/make_model_input.R]: Done.

File: analysis/active_analyses.R
Comment (venexia) [inline analysis/active_analyses.R:72]: Please could you give the values `vascular_risk`, `parkinson_risk`, `cognitive_impairment` and `parkinson_disease` more informative names? Code such as `parkinson_disease <- c("out_date_any_dementia")` is potentially confusing.

File: analysis/active_analyses.R
Comment (venexia) [inline analysis/active_analyses.R:556]: Only `cov_bin_hypertension` and `cov_bin_diabetes` need to be removed from this model as these are the only variables used to define the subgroup.

File: analysis/active_analyses.R
Comment (venexia) [inline analysis/active_analyses.R:533]: Only `cov_bin_hypertension` and `cov_bin_diabetes` need to be removed from this model as these are the only variables used to define the subgroup.

File: analysis/active_analyses.R
Comment (venexia) [inline analysis/active_analyses.R:583]: Please can you remove any comments containing old code so that this is easier to read?

File: analysis/model/make_model_input.R
Comment (venexia) [inline analysis/model/make_model_input.R:82]: `cov_bin_hypertension` and `cov_bin_diabetes` are not needed here as you defined the variable `sub_bin_high_vascular_risk`

**Review by venexia — CHANGES_REQUESTED** [review]
Hi @josephignace - Thank you for your further work on this PR. Just a couple more changes required before approval.

File: analysis/active_analyses.R
Comment (josephignace) [reply analysis/active_analyses.R:544]: Changed to analysis = sub_bin_high_vascular_risk_true. This variable was named in the preprocess (already ran).

File: analysis/active_analyses.R
Comment (josephignace) [reply analysis/active_analyses.R:567]: Changed to analysis = sub_bin_high_vascular_risk_false. This variable was named in the preprocess (already ran).

File: analysis/model/make_model_input.R
Comment (josephignace) [reply analysis/model/make_model_input.R:82]: Done.

File: analysis/active_analyses.R
Comment (josephignace) [reply analysis/active_analyses.R:583]: Done.

File: analysis/active_analyses.R
Comment (josephignace) [reply analysis/active_analyses.R:533]: Done.

File: analysis/active_analyses.R
Comment (josephignace) [reply analysis/active_analyses.R:72]: Added suffix "_sub_out" and a note indicating which covariate was removed.
out_date_cognitive_impairment_symptons cannot be renamed (preprocess already ran). Rename should be made in Stata.

File: analysis/model/make_model_input.R
Comment (venexia) [reply analysis/model/make_model_input.R]: This is because the dummy data is small but for 17mil rows, an extra variable will make more difference.

File: analysis/model/make_model_input.R
Comment (venexia) [inline analysis/model/make_model_input.R:69]: Remove commented out code as not used anymore and in the Git history.

File: analysis/active_analyses.R
Comment (venexia) [inline analysis/active_analyses.R:1]: `outcomes_runmain` could also be removed as no longer used and in the Git history.

Comment (josephignace) [reply analysis/active_analyses.R:1]: Done.

File: analysis/model/make_model_input.R
Comment (josephignace) [reply analysis/model/make_model_input.R:69]: Done.

**Review by venexia — CHANGES_REQUESTED** [review]
Hi @josephignace. Thank you for addressing all my comments. My local testing finished this morning and I have identified a further issue. Using the analysis 'cohort_prevax-main-cognitive_impairment_symptoms' as an example: the list of covariates in the active_analyses table includes the covariate `cov_bin_history_cog_imp_sympt` but this should not be included in the model for this outcome. It has been correctly removed from the model input file but it is still present in the covariate list in the active_analyses table so the code looks for it. Please could you write some code that removes the covariate from the list for the outcome in active_analyses.R. Many thanks.

**Comment by venexia** [comment]
Hi @josephignace. The edits you have made do not address my previous comment, which applies to all analyses (not just subgroups). The example I gave used the main analysis, which has the same covariate list before and after your changes. 

My suggestion for addressing the comment is to add some code to the end of the active_analyses.R script that says "for all analyses where the outcome is X, remove covariate Y from covariate_other". 

Please also revert the changes you made to the subgroup analyses as these were correct before.

**Comment by josephignace** [comment]
Hi @venexia active_analyses updated.

File: analysis/active_analyses.R
Comment (venexia) [inline analysis/active_analyses.R:617]: I think we could simplify this. I am using cognitive impairment symptoms as an example here.

Instead of looping over each row of `df` as shown below:
```
for (i in 1:nrow(df)) {

  analysis <- df$analysis[i]
  outcome <- df$outcome[i]

  if (outcome == "out_date_cognitive_impairment_symptoms") {
    df$covariate_other[i] <- gsub("cov_bin_history_cog_imp_sympt;", "", df$covariate_other[i])
  }

}
```

We could edit all values of `df$covariate_other` based on the value in `df$outcome` at the same time:

```
df$covariate_other <- ifelse(df$outcome=="out_date_cognitive_impairment_symptoms",gsub("cov_bin_history_cog_imp_sympt;", "", df$covariate_other),df$covariate_other)
```

Comment (josephignace) [reply analysis/active_analyses.R:613]: Done.

File: analysis/active_analyses.R
Comment (venexia) [inline analysis/active_analyses.R:634]: This is correctly implemented in terms of code but (using lewy body dementia as an example) we don't have a `cov_bin_history_lewy_body_dementia` in the analysis, just `any dementia`. I think it is best we discuss this at the meeting tomorrow.

File: analysis/active_analyses.R
Comment (venexia) [reply analysis/active_analyses.R:62]: This comment has not been addressed from the previous review: Please remove any mention of covariates from this note. The suffix _sub_out is all about outcomes, we handle the covariates using gsub() - the same as we have always done for subgroups (e.g., age, sex, etc.).

**Review by venexia — CHANGES_REQUESTED** [review]
Hi @josephignace. The code is correctly implemented but we have one outstanding issue that we need to discuss at the meeting tomorrow. There is also a comment from a previous review that has not been addressed. See you then.

File: analysis/active_analyses.R
Comment (josephignace) [reply analysis/active_analyses.R:62]: Done.

File: analysis/active_analyses.R
Comment (josephignace) [reply analysis/active_analyses.R]: Updated.

**Comment by josephignace** [comment]
Hi @venexia Age models added and ready for review.

File: analysis/active_analyses.R
Comment (venexia) [inline analysis/active_analyses.R:237]: 65 is not included in this age category, please rename.

File: analysis/active_analyses.R
Comment (venexia) [inline analysis/active_analyses.R:259]: 85 is not included in this age category, please rename.

**Review by venexia — CHANGES_REQUESTED** [review]
Hi @josephignace. All original changes have now been addressed but the naming of the age models needs to be revisited.

File: analysis/active_analyses.R
Comment (josephignace) [reply analysis/active_analyses.R:237]: Done.

**Pull request merged**
